cosmwasm/packages/wasmi-runtime/src/cosmwasm/types.rs

```diff
@@ -104,27 +105,130 @@ pub struct Coin {
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
-#[serde(rename_all = "lowercase")]
-pub enum CosmosMsg {
+#[serde(untagged)]
+pub enum WasmOutput {
+    ErrObject {
+        #[serde(rename = "Err")]
+        err: Value,
+    },
+    OkString {
+        #[serde(rename = "Ok")]
+        ok: String,
+    },
+    OkObject {
+        #[serde(rename = "Ok")]
+        ok: OkObject,
+    },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct OkObject {
+    pub messages: Vec<CosmosMsg>,
+    pub log: Vec<OutputLog>,
+    pub data: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub struct OutputLog {
+    pub key: String,
+    pub value: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+// See https://github.com/serde-rs/serde/issues/1296 why we cannot add De-Serialize trait bounds to T
+pub enum CosmosMsg<T = CustomMsg>
+where
+    T: Clone + fmt::Debug + PartialEq,
+{
+    Bank(BankMsg),
+    // by default we use RawMsg, but a contract can override that
+    // to call into more app-specific code (whatever they define)
+    Custom(T),
+    Staking(StakingMsg),
+    Wasm(WasmMsg),
+}
```

Comment:
Seems like you changed the definition of `CosmosMsg`. why is that?
After another look - did you basically refresh an old definition with how it looks like upstream now?

Reply:
Yes, keep in mind this is a copy of the original enum from `cosmwasm-std`, and should be in correlation to it.
I will add a comment describing that.

Reply:
Mhm.
I think there already is a comment in the top of the file

Reply:
Oh you're right it is.
I updated it and will keep the specific comments just in case :)